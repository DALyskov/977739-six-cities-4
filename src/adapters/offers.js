export const createOffers = (data) =>
  data.map((offer) => {
    return {
      bedrooms: offer.bedrooms,
      city: offer.city,
      description: offer.description,
      features: offer.goods,

      hostName: offer.host.name,
      hostAvatar: offer.host.avatar_url,
      isHostPro: offer.host.is_pro,
      hostId: offer.host.id,

      id: offer.id,
      images: offer.images,
      isBookmark: offer.is_favorite,
      isPremium: offer.is_premium,

      location: offer.location,

      maxAdults: offer.max_adults,
      previewImg: offer.preview_image,
      price: offer.price,
      rating: offer.rating,
      name: offer.title,
      type: offer.type,
    };
  });
