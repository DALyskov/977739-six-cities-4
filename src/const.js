export const APPROVED_NAME = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];
