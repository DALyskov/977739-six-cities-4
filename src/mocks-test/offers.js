export const offers = [
  {
    bedrooms: 1,
    city: {
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
      name: `Amsterdam`,
    },
    description: `I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.`,
    features: [
      `Coffee machine`,
      `Laptop friendly workspace`,
      `Air conditioning`,
      `Baby seat`,
      `Fridge`,
      `Breakfast`,
      `Dishwasher`,
      `Towels`,
      `Washer`,
      `Washing machine`,
    ],

    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: true,
    hostId: 25,

    id: 5,
    images: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    isBookmark: false,
    isPremium: false,

    location: {
      latitude: 52.397540000000006,
      longitude: 4.9099759999999995,
      zoom: 16,
    },

    maxAdults: 2,
    previewImg: `img/room.jpg`,
    price: 101,
    rating: 3.5,
    name: `Perfectly located Castro`,
    type: `room`,
  },

  {
    bedrooms: 5,
    city: {
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
      name: `Amsterdam`,
    },
    description: `I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.`,
    features: [
      `Coffee machine`,
      `Laptop friendly workspace`,
      `Air conditioning`,
      `Baby seat`,
      `Fridge`,
      `Breakfast`,
      `Dishwasher`,
      `Towels`,
      `Washer`,
      `Washing machine`,
    ],

    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: true,
    hostId: 25,

    id: 1,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    isBookmark: false,
    isPremium: false,

    location: {
      latitude: 52.397540000000006,
      longitude: 4.886976,
      zoom: 16,
    },

    maxAdults: 6,
    previewImg: `img/room.jpg`,
    price: 813,
    rating: 2.4,
    name: `Wood and stone place`,
    type: `house`,
  },

  {
    bedrooms: 4,
    city: {
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
      name: `Amsterdam`,
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    features: [`Breakfast`, `Laptop friendly workspace`],

    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: true,
    hostId: 25,

    id: 14,
    images: [`img/apartment-01.jpg`, `img/apartment-03.jpg`],
    isBookmark: false,
    isPremium: true,

    location: {
      latitude: 52.397540000000006,
      longitude: 4.886976,
      zoom: 16,
    },

    maxAdults: 7,
    previewImg: `img/room.jpg`,
    price: 423,
    rating: 4.7,
    name: `Penthouse, 4-5 rooms + 5 balconies`,
    type: `hotel`,
  },
];

export const originOffers = [
  {
    bedrooms: 1,
    city: {
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
      name: `Amsterdam`,
    },
    description: `I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.`,
    goods: [
      `Coffee machine`,
      `Laptop friendly workspace`,
      `Air conditioning`,
      `Baby seat`,
      `Fridge`,
      `Breakfast`,
      `Dishwasher`,
      `Towels`,
      `Washer`,
      `Washing machine`,
    ],

    host: {
      'avatar_url': 'img/avatar-angelina.jpg',
      id: 25,
      'is_pro': true,
      name: 'Angelina',
    },
    id: 5,
    images: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    'is_favorite': false,
    'is_premium': false,
    location: {
      latitude: 52.397540000000006,
      longitude: 4.9099759999999995,
      zoom: 16,
    },
    'max_adults': 2,
    'preview_image': `img/room.jpg`,
    price: 101,
    rating: 3.5,
    title: `Perfectly located Castro`,
    type: `room`,
  },
];
