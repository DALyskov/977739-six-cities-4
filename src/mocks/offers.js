export default [
  {
    id: 1,
    isPremium: true,
    images: [
      `img/apartment-01.jpg`,
      `img/room.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    price: 120,
    isBookmark: false,
    rating: 4.5,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    bedrooms: 3,
    maxAdults: 1,
    features: [
      'Heating',
      'Kitchen',
      'Cable TV',
      'Washing machine',
      'Coffee machine',
    ],
    hostName: `Dima`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: true,
    description:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: 'Amsterdam',
    },
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 8,
    },
  },
  {
    id: 2,
    isPremium: false,
    images: [`img/room.jpg`],
    price: 80,
    isBookmark: true,
    rating: 4,
    name: `Wood and stone place`,
    type: `Private room`,
    bedrooms: 2,
    maxAdults: 3,
    features: [
      'Heating',
      'Kitchen',
      'Cable TV',
      'Washing machine',
      'Coffee machine',
      'Dishwasher',
    ],
    hostName: `Alex`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: false,
    description:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: 'Amsterdam',
    },
    location: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198,
      zoom: 8,
    },
  },
  {
    id: 3,
    isPremium: false,
    images: [`img/apartment-02.jpg`],
    price: 132,
    isBookmark: false,
    rating: 4,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    bedrooms: 5,
    maxAdults: 2,
    features: ['Heating', 'Cable TV', 'Coffee machine', 'Dishwasher'],
    hostName: `Pavel`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: false,
    description:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: 'Amsterdam',
    },
    location: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198,
      zoom: 8,
    },
  },
  {
    id: 5,
    isPremium: true,
    images: [
      `img/apartment-01.jpg`,
      `img/room.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    price: 120,
    isBookmark: false,
    rating: 4.5,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    bedrooms: 3,
    maxAdults: 1,
    features: [
      'Heating',
      'Kitchen',
      'Cable TV',
      'Washing machine',
      'Coffee machine',
    ],
    hostName: `Dima`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: true,
    description:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    city: {
      location: {
        latitude: 48.8652777777778,
        longitude: 2.3486111111111,
        zoom: 10,
      },
      name: 'Paris',
    },
    location: {
      latitude: 48.8653777777778,
      longitude: 2.3486111111111,
      zoom: 8,
    },
  },
  {
    id: 4,
    isPremium: true,
    images: [`img/apartment-03.jpg`],
    price: 180,
    isBookmark: false,
    rating: 5,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    bedrooms: 1,
    maxAdults: 4,
    features: ['Kitchen', 'Washing machine', 'Coffee machine', 'Dishwasher'],
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: true,
    description:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: 'Amsterdam',
    },
    location: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198,
      zoom: 8,
    },
  },
];
