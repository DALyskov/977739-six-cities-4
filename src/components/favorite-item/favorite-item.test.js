import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import {offers} from '../../mocks-test/offers.js';
import {SORTING_ITEMS} from '../../const.js';

import history from '../../history.js';
import FavoriteItem from './favorite-item.jsx';
import NameSpace from '../../reducer/name-space.js';

const mockStore = configureStore([]);

describe(`FavoriteItem_snapchots`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      favoriteOffers: offers,
    },
    [NameSpace.STATE_APPLICATION]: {
      sortingType: SORTING_ITEMS[0],
    },
    [NameSpace.USER]: {},
  });
  it(`FavoriteItem_with_data`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router history={history}>
            <FavoriteItem
              favoriteOffers={offers}
              favoriteCity={offers[0].city.name}
              onCityClick={() => {}}
            />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
