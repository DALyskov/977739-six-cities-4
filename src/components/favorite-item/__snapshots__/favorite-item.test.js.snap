// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteItem_snapchots FavoriteItem_with_data 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="/"
        onClick={[Function]}
      >
        <span>
          Amsterdam
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <article
      className="favorites__card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={110}
            src="img/apartment-01.jpg"
            width={150}
          />
        </a>
      </div>
      <div
        className="favorites__card-info place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              101
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            className="place-card__link"
            href="/offer/5"
            onClick={[Function]}
          >
            Perfectly located Castro
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          room
        </p>
      </div>
    </article>
    <article
      className="favorites__card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={110}
            src="img/apartment-01.jpg"
            width={150}
          />
        </a>
      </div>
      <div
        className="favorites__card-info place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              813
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "40%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            className="place-card__link"
            href="/offer/1"
            onClick={[Function]}
          >
            Wood and stone place
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          house
        </p>
      </div>
    </article>
    <article
      className="favorites__card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={110}
            src="img/apartment-01.jpg"
            width={150}
          />
        </a>
      </div>
      <div
        className="favorites__card-info place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              423
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            className="place-card__link"
            href="/offer/14"
            onClick={[Function]}
          >
            Penthouse, 4-5 rooms + 5 balconies
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          hotel
        </p>
      </div>
    </article>
  </div>
</li>
`;
